### PR TITLE
[ROCm] add support for HIP stream priorities

### PR DESCRIPTION
**Pull request by jeffdaily**

### CODE REVIEW

**Comment by vincentqb** [comment]
@VitalyFedyunin -- do you want to review this PR?

File: torch/cuda/streams.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ class Stream(torch._C._CudaStreamBase):
                                  represent higher priorities.
     """
 
-    def __new__(cls, device=None, priority=0, **kwargs):
+    def __new__(cls, device=None, priority=0 if torch.version.cuda else 1, **kwargs):
```
Comment (ezyang) [inline torch/cuda/streams.py:20]: This is seems bad. Why do you have to do this?

**Review by ezyang — CHANGES_REQUESTED** [review]
I think I would prefer if priorities had the same semantics across CUDA and HIP in the frontend, and HIP just remaps the priorities to -1 or 0 accordingly.

File: torch/cuda/streams.py (hunk 1 above)
Comment (jeffdaily) [reply torch/cuda/streams.py:20]: Following CUDA semantics, priority 0 is default.  There isn't any way to distinguish between a caller not specifying a priority (defaulting to 0 i.e. default priority), and a ROCm user having specifically asked for 0==high priority after a call to `priority_range()`.

Comment (ezyang) [reply torch/cuda/streams.py:20]: OK. Then my broader question is, how come the code for ROCm has to be different than the code for CUDA? Are ROCm and CUDA different, or is the claim that the CUDA behavior is bad and the ROCm version of this behavior is better?

**Comment by jeffdaily** [comment]
HIP stream priorities were aligned to CUDA conventions in ROCm 3.7.  We will validate this feature first in our fork and internal QA processes and then issue a new PR.

**Pull request closed without merging**